src/Auth/EmailVerification/EmailVerificationRepository.php

```diff
@@ -0,0 +1,206 @@
+            'senderName' => (config('nodes.api.email-verification.from.name') != 'Nodes') ? config('nodes.api.email-verification.from.name') : config('nodes.project.name')
+        ], function($message) use ($user) {
+            $message->to($user->email)
+                ->from(config('nodes.api.remail-verification.from.email', 'no-reply@nodes.dk'), config('nodes.api.email-verification.from.name', 'Nodes'))
+                ->subject(config('nodes.api.email-verification.subject', 'Email verification request'));
+        });
+
+        return $status;
+    }
+
+    /**
+     * Generate reset token
+     *
+     * @author Morten Rugaard <moru@nodes.dk>
+     *
+     * @access protected
+     * @param  \Illuminate\Database\Eloquent\Model $user
+     * @return string
+     */
+    protected function generateEmailVerificationToken(IlluminateModel $user)
+    {
+        // Generate new token using Laravel's encryption key
+        $token = hash_hmac('sha256', str_random(40), config('app.key'));
+
+        // Expire timestamp
+        $expire = Carbon::now()->addMinutes(config('nodes.api.reset-password.expire', 120));
+
+        // If user has previously tried to reset his/her password
+        // we should just update the token of the previous entry
+        // instead of creating a new one
+        $verificationToken = $this->where('email', '=', $user->email)->first();
+        if (!empty($verificationToken)) {
+            $verificationToken->update(['token' => $token, 'used' => 0, 'expire_at' => $expire->format('Y-m-d H:i:s')]);
+        } else {
+            $this->insert(['email' => $user->email, 'token' => $token, 'expire_at' => $expire->format('Y-m-d H:i:s')]);
+        }
+
+        return $token;
+    }
+
+    /**
+     * @param $email
+     * @return bool
+     * @author Paulius Navickas <pana@nodes.dk>
+     */
+    public function updateVerificationByEmail($email)
+    {
+        // Retrieve user by e-mail
+        $user = $this->authModel->where('email', '=', $email)->first();
+        if (empty($user)) {
+            return false;
+        }
+
+        // Update user with new password
+        return (bool) $user->update([
+            'verified_at' => Carbon::now()
+        ]);
+    }
+
+
+    /**
+     * @param $email
+     * @param $token
+     * @return bool
+     * @author Paulius Navickas <pana@nodes.dk>
+     */
+    public function verify($email, $token)
+    {
+        $user = $this->with('verification')
+            ->where('email', $email)
+            ->first();
+
+
+        // E-mail does not exist
+        if (empty($user) || !$user->verification instanceof EmailVerificationModel) {
+            return false;
+        }
+
+        if($user->verification->user){
+            return false;
+        }
+
+        // Validate password
+        if (strcmp($token, $user->verification->token) !== 0) {
+            return false;
+        }
+        $user->verified_at = new Carbon();
```

Comment:
I would recommend using the `user->fill()` method instead of assigning them directly on the object. This way it goes throw the "Laravel loop" like marking the model as dirty, seeing if the fields are fillable etc.